Lib/test/test_xmlrpc.py

```diff
@@ -343,77 +343,71 @@ def run_server():
             self.assertEqual(p.method(), 5)
             self.assertEqual(p.method(), 5)
 
-    def test_simple_xml_rpc_dispatcher_resolves_registered_func_and_does_not_chain_excs_when_calling_it(self):
-        exp_params = 1, 2, 3
-        dispatched = False
 
-        class DispatchExc(Exception):
-            """Raised inside the dispatched func"""
+class SimpleXMLRPCDispatcherTestCase(unittest.TestCase):
+    class DispatchExc(Exception):
+        """Raised inside the dispatched functions when checking for chained exceptions"""
+
+        def __init__(self, *args, method=None, params=None, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.method = method
+            self.params = params
+
+    def test_call_registered_func(self):
+        """Attempts to resolve an explicitly registered function and call it, making sure any exception raised
+        inside the function has no other exception chained to it"""
+
+        exp_params = 1, 2, 3
 
         def dispatched_func(*params):
-            nonlocal dispatched
-            dispatched = True
-            self.assertEqual(params, exp_params)
-            raise DispatchExc()
+            raise SimpleXMLRPCDispatcherTestCase.DispatchExc(params=params)
 
         dispatcher = xmlrpc.server.SimpleXMLRPCDispatcher()
         dispatcher.register_function(dispatched_func)
-        try:
-            dispatcher._dispatch(method='dispatched_func', params=exp_params)
-        except DispatchExc as e:
-            self.assertFalse(e.__cause__)
-            self.assertFalse(e.__context__)
-        self.assertTrue(dispatched)
+        with self.assertRaises(SimpleXMLRPCDispatcherTestCase.DispatchExc) as exc_ctx:
```

Comment:
I used `self` for now. I don't want to define it at module level just yet. There is nothing else using it. That kind of refactoring, in my book, should happen when somebody introduces another class (test case) that would use it